src/otx/core/data/dataset/base.py

```diff
@@ -0,0 +1,80 @@
+"""Base class for OTXDataset."""
+from __future__ import annotations
+
+from abc import abstractmethod
+from collections.abc import Iterable
+from typing import TYPE_CHECKING, Callable, Generic, List, Optional, Union
+
+import numpy as np
+from torch.utils.data import Dataset
+
+from otx.core.data.entity.base import T_OTXDataEntity
+
+if TYPE_CHECKING:
+    from datumaro import DatasetSubset
+
+Transforms = Union[Callable, List[Callable]]
```

Comment:
Is the type alias `Transforms = Union[Callable, List[Callable]]` essential? What would be the issue with directly using `Union[Callable, List[Callable]]`? When encountering this `Transforms` type hint, how can one discern that it represents a `Union[Callable, List[Callable]]`?

Reply:
What's your alternative? Do you want just copy and paste `Union[Callable, List[Callable]]` everywhere in the codebase?

Reply:
There seems to be an inconsistency. In src/otx/core/data/transform_libs/mmcv.py, transforms are specified as List[Callable]. However, here it's defined as a union of Callable or List[Callable]. I would recommend standardizing it to List[Callable] and eliminating the Transforms type alias for clarity.